Sources/Basic/JSON.swift

```diff
@@ -122,23 +126,22 @@ extension JSON: ByteStreamable {
         case .string(let value):
             stream <<< Format.asJSON(value)
         case .array(let contents):
-            // FIXME: OutputByteStream should just let us do this via conformances.
-            stream <<< "["
+            stream <<< "[" <<< "\n"
             for (i, item) in contents.enumerated() {
-                if i != 0 { stream <<< ", " }
-                stream <<< item
+                if i != 0 { stream <<< "," <<< "\n" }
+                stream <<< " ".repeating(n: indent + 2)
+                item.write(to: stream, indent: indent + 2)
             }
-            stream <<< "]"
+            stream <<< "\n" <<< " ".repeating(n: indent) <<< "]"
```

Comment:
We should get an OutputByteStream way to write a repeating sequence -- this way is inefficient as it creates the string just to throw it away.

The simple way to do this is to add a `Format` style wrapper type for writing a repeating character.